core/src/main/scala/kafka/log/Log.scala

```diff
@@ -2131,15 +2139,21 @@ class Log(@volatile var dir: File,
    *
    * @throws IOException if the file can't be renamed and still exists
    */
-  private def asyncDeleteSegment(segment: LogSegment) {
+  private def deleteSegmentFiles(segment: LogSegment, asyncDelete: Boolean = true) {
```

Comment:
perhaps rename to deleteSegment ? also its singular ?

Reply:
I wanted to make a distinction between `removeAndDeleteSegment` which performs a logical deletion and schedules a physical deletion vs this method which only performs physical deletion. Each segment is made of multiple files and thus `deleteSegmentFiles`. If this is confusing, I am open to ideas for a better name.

Reply:
makes sense, thanks

Reply:
I looked the code I see other related functions like mayDeleteSegment, deleteSegments, deleteOldSegments, it feels that renaming by adding "Files" throws the consistency away, calling it just deleteSegment makes sense to me

Reply:
I am not sure I follow. Are you suggesting the distinction between physical and logical deletion is not worth having? Note that a logical deletion eventually causes a physical deletion, so all the methods you pointed to would eventually call `deleteSegmentFiles`. I don't feel very strongly about the naming so I could change it back but I disagree that this throws the consistency away.

Reply:
i agree with you, it is better to have distinction, thanks for the explanation :)

Reply:
I'd suggest not having any default for `asyncDelete` to avoid unintentional bugs.

Reply:
> I'd suggest not having any default for asyncDelete to avoid unintentional bugs.

Sounds good. I made the change in 509b916.